### DISCUSSION TITLE
Cell decorator unexpectedly treats single value list the same as single value

### DISCUSSION THREAD

**Discussion by jonnyroth** [gdsfactory/gdsfactory#2750] [Q&A] [answered]
Hi,
I've encountered unexpected behavior with the cell decorator from gdsfactory when passing a single-value list as an argument to a decorated function. My function is designed to accept either an integer or a list of integers. If I pass a single integer, say 5, the function should create 5 instances of a geometry. However, if I pass a list like [0, 1, 4], it should create several instances while keeping the same origin.

The unexpected behavior occurs when I pass a single-value list, such as [5], intended to create one instance that is not at the origin. It seems that the cell decorator is treating [5] the same as 5, which is not the intended behavior.

Here is a code example that illustrates the issue:

```python
import gdsfactory as gf

@gf.cell
def single_value_list_test(in_variable: int | list) -> gf.Component:
    print(in_variable)
    return gf.Component()

  single_value_list_test([1,2,3]) #prints input variable
  single_value_list_test([1,2,3]) #does not print, already cached
  single_value_list_test([4]) #prints input variable as list
  single_value_list_test(4) #does not print, treats as duplicate of previous cached
  single_value_list_test(3) #prints input variable as int
  single_value_list_test([3]) #does not print, treats as duplicate of previous cached
```
Is this behavior intentional? If not, should the cell decorator ideally treat [5] and 5 differently? I can work around this by defining two separate parameters for list and int input, and using only one at a time, but I wanted to highlight this in case it's an issue worth addressing.

Environment details:
Python version: 3.10
gdsfactory version: 7.23.0

Thank you for your help!

**Comment by joamatab** [comment, marked as answer]
Hi Jonny,

yes, we could fix, the issue is that a list with one value is serializing to the same as a single value

**Reply by jonnyroth** [reply]
Thanks @joamatab.  I will leave it up to you whether it's worth the effort, since it's not too much trouble for me to work around the issue.
